core/flyout_separator.ts

```diff
@@ -0,0 +1,17 @@
+import type {IBoundedElement} from './interfaces/i_bounded_element.js';
+import {Rect} from './utils/rect.js';
+
+export class FlyoutSeparator implements IBoundedElement {
+  private x = 0;
+  private y = 0;
+  constructor(private gap: number) {}
+
+  getBoundingRectangle(): Rect {
```

Comment:
If someone uses a really big gap value in a vertical flyout, can it cause the flyout to become excessively wide as a result of the gap being a square instead of a rectangle? If yes, I think we should have `gapX` and `gapY` be specified separately

Reply:
Yes; updated it to take an axis argument and be zero-width on the non-dominant axis.